test/elixir_internal_certificate/account/accounts_test.exs

```diff
@@ -0,0 +1,204 @@
+defmodule ElixirInternalCertificate.Account.AccountsTest do
+  use ElixirInternalCertificate.DataCase
+
+  alias ElixirInternalCertificate.Account.Accounts
+  alias ElixirInternalCertificate.Account.Schemas.{User, UserToken}
+
+  describe "get_user_by_email/1" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when the email does not exist, does not return the user" do
+      assert Accounts.get_user_by_email("unknown@example.com") == nil
+    end
+
+    test "with the existed email, returns the user", %{user: user} do
+      %{id: id} = user
+
+      assert %User{id: ^id} = Accounts.get_user_by_email(user.email)
+    end
+  end
+
+  describe "get_user_by_email_and_password/2" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when the email does not exist, does not return the user" do
+      assert Accounts.get_user_by_email_and_password("unknown@example.com", "hello world!") == nil
+    end
+
+    test "when the password is not valid, does not return the user", %{user: user} do
+      assert Accounts.get_user_by_email_and_password(user.email, "invalid") == nil
+    end
+
+    test "when the email and password are valid, returns the user", %{user: user} do
+      %{id: id} = user
+
+      assert %User{id: ^id} =
+               Accounts.get_user_by_email_and_password(user.email, valid_user_password())
+    end
+  end
+
+  describe "get_user!/1" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when id is invalid, raises a no result error" do
+      assert_raise Ecto.NoResultsError, fn ->
+        Accounts.get_user!(-1)
+      end
+    end
+
+    test "with the given id, returns the user", %{user: user} do
+      %{id: id} = user
+
+      assert %User{id: ^id} = Accounts.get_user!(user.id)
+    end
+  end
+
+  describe "register_user/1" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when email and password are blank, requires validation error will be throw" do
+      {:error, changeset} = Accounts.register_user(%{})
```

Comment:
same as above situation with the `setup hook` in this test, do we really need to create `a user in the database` :D